src/main/webapp/script.js

```diff
@@ -0,0 +1,100 @@
+var zoomedIn = false;
+var aqLayerOn = true;
+var routeVisible = false;
+
+function changeBackgroundImage(imagePath) {
+    document.body.style.backgroundImage = "url('" + imagePath + "')";
+}
+
+function getImagePath() {
+    path = "";
+    if (zoomedIn) {
+        path = "images/wagga_";
+    } else {
+        path = "images/nsw_";
+    }
+
+    if (aqLayerOn) {
+        path += "gradient";
+    } else {
+        path += "plain";
+    }
+
+    if (routeVisible) {
+        path += "_route.png";
+    } else {
+        path += ".png";
+    }
+
+    return path;
+}
+
+function toggleAQLayer() {
+    checkbox = document.getElementById("aq-layer-toggle");
+    
+    aqLayerOn = checkbox.checked;
```

Comment:
The variables that track this state in the script seem like unnecessary caching. Consider dropping the variables, and having getImagePath() directly query the HTML state?

Reply:
Do you mean querying what the image currently is?

Reply:
As a side note, changing the background image isn't the final implementation for the display. The plan is to use a maps widget from the maps API.

Reply:
Sorry; to be more specific; you query and store the state of the checkbox:
```
aqLayerOn = checkbox.checked;
```
And caches the state in a global variable; which is then used by the next function. The global isn't required because the next function can just as easily check the state of the element itself.

The pattern of storing state in a global is valid; but should be used for tracking more complex things that aren't readily available; for example, you might use it to cache AQ data that's been pulled in from one request, to reuse in subsequent requests if it overlaps and is still "fresh".